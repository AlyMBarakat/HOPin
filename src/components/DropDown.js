import React, { useState } from 'react';
import {
    Text,
    View,
    StyleSheet,
    TouchableOpacity,

} from 'react-native';
import { Picker } from '@react-native-picker/picker';

const DropDown = ({ list, name, selectedItem, setSelectedItem }) => {
    const [isListShown, setListShown] = useState(false);

    return (
        <View style={{ marginBottom: 22, }}>
            <Text style={styles.inputHeader}>{name}</Text>
            <TouchableOpacity
                style={styles.input}
                onPress={() => setListShown(isListShown => !isListShown)}
            >
                <Text style={{ color: '#2D3436' }}>{selectedItem}</Text>
            </TouchableOpacity>
            {
                isListShown &&
                <Picker
                    selectedValue={selectedItem}
                    onValueChange={(itemValue, itemIndex) =>
                        setSelectedItem(itemValue)
                    }
                    itemStyle={{ color: '#fff' }}
                >
                    {
                        list.map((item, index) => {
                            return <Picker.Item key={item} label={item} value={item} />
                        })
                    }
                </Picker>
            }
        </View>
    );
};


const styles = StyleSheet.create({
    inputHeader: {
        color: "#fff",
        fontSize: 15,
        fontWeight: 'bold',
        alignSelf: 'flex-start',
        marginBottom: 12,
    },
    input: {
        backgroundColor: '#72B5A8',
        borderWidth: 3,
        borderRadius: 5,
        borderColor: '#3C796D',
        paddingHorizontal: 6,
        paddingVertical: 10,
    },
})

export default DropDown;
