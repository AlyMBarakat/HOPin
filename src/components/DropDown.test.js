import React from 'react';
import renderer from 'react-test-renderer';
import DropDown from './DropDown';

const testList = ['temp1', 'temp2', 'temp3', 'temp4', 'temp5', 'temp6'];

test('renders dropdown picker correctly', () => {
  const tree = renderer
    .create(<DropDown list={testList} name="Dropdown list" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
