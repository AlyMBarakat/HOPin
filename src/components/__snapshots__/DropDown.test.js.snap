// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders date and time pickers correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 22,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "#fff",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginBottom": 12,
      }
    }
  >
    Dropdown list
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#72B5A8",
        "borderColor": "#3C796D",
        "borderRadius": 5,
        "borderWidth": 3,
        "opacity": 1,
        "paddingHorizontal": 6,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2D3436",
        }
      }
    />
  </View>
</View>
`;

exports[`renders dropdown picker correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 22,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "flex-start",
        "color": "#fff",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginBottom": 12,
      }
    }
  >
    Dropdown list
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#72B5A8",
        "borderColor": "#3C796D",
        "borderRadius": 5,
        "borderWidth": 3,
        "opacity": 1,
        "paddingHorizontal": 6,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2D3436",
        }
      }
    />
  </View>
</View>
`;
